gst-plugin-pravega/src/pravegasink/imp.rs

```diff
@@ -73,11 +80,135 @@ pub enum TimestampMode {
     Tai = 2,
 }
 
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Clone, Copy, glib::GEnum)]
+#[repr(u32)]
+#[genum(type_name = "GstRetentionType")]
+pub enum RetentionType {
+    #[genum(
+        name = "If 'none', no data will be deleted from the stream. ",
+        nick = "none"
+    )]
+    None = 0,
+    #[genum(
+        name = "If 'days', data older than 'retention-days' will be deleted from the stream.",
+        nick = "days"
+    )]
+    Days = 1,
+    #[genum(
+        name = "If 'bytes', the oldest data will be deleted so that the data size does not exceed 'retention-bytes'.",
+        nick = "bytes"
+    )]
+    Bytes = 2,
+    #[genum(
+        name = "If 'daysAndBytes', the oldest data will be deleted if it is older than 'retention-days' or the data size exceeds 'retention-bytes'.",
+        nick = "daysAndBytes"
+    )]
+    DaysAndBytes = 3,
+}
+
+#[derive(Debug)]
+enum RetentionPolicy {
+    Days(f64),
+    Bytes(u64),
+    DaysAndBytes(f64, u64),
+    None,
+}
+
+impl RetentionPolicy {
+    fn new(retention_type: RetentionType, days: Option<f64>, bytes: Option<u64>) -> Self {
+        match retention_type {
+            RetentionType::Days => Self::Days(days.expect("retention-days is not set")),
```

Comment:
This would cause a panic if the user forgets to specify retention-days. It would be better to have `new()` return an `anyhow::Result<Self>` with an error and then have `Start` log the error.

Reply:
changed. Please take a check again.